src/quartz/context/context.cpp

```diff
@@ -24,6 +25,15 @@ Context::Context(const std::vector<GateType> &supported_gates,
       supported_quantum_gates_.emplace_back(gate);
     }
   }
+
+  // Precomputes whether any of the gates use halved parameters.
+  may_use_halved_params_ = false;
+  for (const auto &g : gates_) {
+    for (int i = 0; i < g.second->get_num_parameters(); ++i) {
+      bool is_halved = g.second->is_param_halved(i);
+      may_use_halved_params_ = may_use_halved_params_ || is_halved;
```

Comment:
Oh sorry, I just realized there is no `||=` operator in C++. I might write something like this but feel free to leave it as is.
```suggestion
      if (g.second->is_param_halved(i)) {
        may_use_halved_params_ = true;
      }
```